### DISCUSSION TITLE
A bug of fullscreen and WebGPU?

### DISCUSSION THREAD

**Discussion by SetoKaiba** [emscripten-core/emscripten#17514] [Q&A]
When I use the WebGPU basic rendering, I click the fullscreen button. The canvas is set to zero width and height.
https://github.com/emscripten-core/emscripten/blob/77da62969eb0b5efe4ce4a0c9ce67c45b9e32638/src/library_browser.js#L409
When the full screen button is clicked, this function is called. And it will call into updateCanvasDimensions.
https://github.com/emscripten-core/emscripten/blob/77da62969eb0b5efe4ce4a0c9ce67c45b9e32638/src/library_browser.js#L732-L777
wNative and hNative is not set.  They will be set to the canvas.widthNative and canvas.heightNative. But there're no widthNative and heightNative as well.
I think the problem is canvas.widthNative and canvas.heightNative is never set.

I can make it working by add this line.
```emscripten_set_canvas_size(kWidth,kHeight);```
But the `emscripten_set_canvas_size` is deprecated. If I use the `emscripten_set_canvas_element_size`, it doesn't set the canvas.widthNative and canvas.heightNative. Then the fullscreen is not working.

xlib, glut, sdl, glfw will all call Browser.setCanvasSize. But the webgpu just get a `SurfaceDescriptorFromCanvasHTMLSelector `. So it doesn't need the initialize with library_xxx.js. Then the canvasSize is never set.

Is it a bug? Should I submit a bug report?

**Comment by sbc100** [comment]
@kainino0x

**Comment by kainino0x** [comment]
This sounds like a bug, unfortunately despite being one of the maintainers of the WebGPU bindings in Emscripten (cc @shrekshao) I'm not familiar enough with Emscripten's canvas code to know how to do this correctly. Please go ahead and file a bug.

**Comment by SetoKaiba** [comment]
https://github.com/emscripten-core/emscripten/issues/17522
@kainino0x I created an issue. But it seems no one repies.
